go/private/rules/test.bzl

```diff
@@ -68,13 +68,19 @@ def _go_test_impl(ctx):
     )
 
     validation_outputs = []
+    nogo_fix_outputs = []
 
     # Compile the library to test with internal white box tests
     internal_library = go.new_library(go, testfilter = "exclude")
     internal_source = go.library_to_source(go, ctx.attr, internal_library, ctx.coverage_instrumented())
     internal_archive = go.archive(go, internal_source)
     if internal_archive.data._validation_output:
         validation_outputs.append(internal_archive.data._validation_output)
+    if internal_archive.data._out_nogo_fix:
+        # Note we do not include external_archive.data._out_nogo_fix since
+        # they correspond to the packages external to the current workspace.
```

Comment:
The distinction between `internal` and `external` archive refers to same package vs. separate package tests here. Could you fix the comment or explain why it's accurate?

Reply:
thanks, 

currently, we retrieve fix via --output_groups=nogo_fix, which is only for the target being built, not including those dependent targets.

thinking about alternatives where we dump fixes for the dependent targets too. maybe by including the fix into the validation action.

Reply:
addressed in the new version.

now, we wire the fix as the output of validation action. This way, --run_validations ensure every target (not only the input target, but also the dependent target) has their fixes if any produced.

The previous implementation is incorrect, since it wires the fix as the outputgroup of the input target only, as a result, we cannot get the fix for the dependent target even if they are linted.

Reply:
back to your original question, still think we should not include external package, given external package is validated independently during which nogo fixes are already produced.